doc/dhcp_server/port_based_dhcp_server_high_level_design.md

```diff
@@ -0,0 +1,1045 @@
+        - [Generate Config](#generate-config)
+        - [Update Lease](#update-lease)
+    - [Customize DHCP Packet Options](#customize-dhcp-packet-options)
+    - [DHCP Relay Daemon](#dhcp-relay-daemon)
+    - [DB Changes](#db-changes)
+        - [Config DB](#config-db)
+            - [DB Objects](#db-objects)
+            - [Yang Model](#yang-model)
+        - [State DB](#state-db)
+            - [DB Objects](#db-objects)
+            - [Yang Model](#yang-model)
+    - [Flow Diagrams](#flow-diagrams)
+        - [DHCP Server Flow](#dhcp-server-flow)
+        - [Config Change Flow](#config-change-flow)
+        - [Lease Update Flow](#lease-update-flow)
+        - [Start flow of dhcprelayd](#start-flow-of-dhcprelayd)
+        - [Work flow of dhcprelayd capture db change](#work-flow-of-dhcprelayd-capture-db-change)
+- [CLI](#cli)
+    - [Config CLI](#config-cli)
+    - [Show CLI](#show-cli)
+- [Test](#test)
+    - [Unit Test](#unit-test)
+        - [Config CLI](#config-cli)
+        - [Show CLI](#show-cli)
+        - [Others](#others)
+    - [Test Plan](#test-plan)
+
+<!-- /TOC -->
+
+
+# Revision
+
+| Rev |     Date    |       Author       | Change Description                  |
+|:---:|:-----------:|:-------------------|:-----------------------------------|
+| 0.1 |  2023/02/08 | Yaqiang Zhu, <br> Jing Kan       | Initial version                     |
+
+# About this Manual
+
+This document describes the design details of **ipv4 port based DHCP server** feature.
+Dynamic Host Configuration Protocol (DHCP) server is used to centrally manage and configure IP addresses of clients dynamically. When server receives requests from DHCP-enabled clients, it would offer information to client about IP address, subnet mask, gateway etc.
+
+# Scope
+This document describes the high level design details about how **ipv4 port based DHCP server** works.
+
+# Definitions/Abbreviations
+###### Table 1: Abbreviations
+| Abbreviation             | Full form                        |
+|--------------------------|----------------------------------|
+| DHCP                      | Dynamic Host Configuration Protocol      |
+
+###### Table 2: Definitions
+| Definitions             | Description                        |
+|--------------------------|----------------------------------|
+| kea-dhcp-server    | Open source DHCP server process distributed by ISC      |
+| dhcrelay | Open source DHCP relay process distributed by ISC |
+
+# Overview
+A DHCP Server is a server on network that can automatically provide and assign IP addresses, default gateways and other network parameters to client devices. 
+
+## Background
+We plan to implement built-in DHCP Server in SONiC to assign IPs based on interface index.
+
+Port-based DHCP Server has below advantages:
+
+1. Relatively more secure and stable, if other no-configured interface requests IP, it will not be assigned.
+2. Assigning IP based on port can quickly complete a self-made network without external information input.
+
```

Comment:
Please add a out of scope section as well to call out specific cases like "MCLAG" or any other , thanks.

Reply:
Added `Design considerations` section